spyderlib/restart_app.py

```diff
@@ -129,18 +181,46 @@ def main():
     shell = os.name != 'nt'
 
     # Wait for original process to end before launching the new instance
-    while True:
+    for counter in range(60*5/SLEEP_TIME):  # Number in seconds (60*5)
         if not is_pid_running(pid):
             break
-        time.sleep(0.2)  # Throttling control
+        time.sleep(SLEEP_TIME)  # Throttling control
+    else:
+        # The old spyder instance took too long to close and restart aborts
+        launch_error_message(type_=CLOSE_ERROR)
 
     env = os.environ.copy()
+
+    # Reset spyder if needed
+    # -------------------------------------------------------------------------
+    if reset:
+        command_reset = '"{0}" "{1}" {2}'.format(python, spyder, args_reset)
+
+        try:
+            p = subprocess.Popen(command_reset, shell=shell)
+            p.communicate()
+            pid_reset = p.pid
+        except Exception as error:
+            p.kill()
+            launch_error_message(type_=RESET_ERROR, error=error)
+
+        # Wait for reset process to end before restarting
+        for counter in range(60/SLEEP_TIME):  # Number in seconds (60)
+            if not is_pid_running(pid_reset):

```

Comment:
If there was an exception before, `pid_reset` won't be defined.
I think this block should go in an `else:` statement after the `try:` and `except:`, since you shouldn''t have to wait if there was an error (I guess).